sources/api/netdog/src/wicked/mod.rs

```diff
@@ -37,6 +37,46 @@ pub(crate) struct WickedInterface {
     pub(crate) ipv6_static: Option<WickedStaticAddress>,
 }
 
+#[derive(Debug, Serialize, PartialEq)]
+pub(crate) struct WickedName {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    namespace: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "$value")]
+    name_body: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "$unflatten=permanent_address")]
+    permanent_address: Option<MacAddress>,
+}
+
+impl WickedName {
+    pub(crate) fn new(id: InterfaceId) -> Self {
+        match id {
+            InterfaceId::Name(name) => Self {
+                namespace: None,
+                name_body: Some(name.to_string()),
+                permanent_address: None,
+            },
+            InterfaceId::MacAddress(mac) => Self {
+                namespace: Some("ethernet".to_string()),
+                name_body: None,
+                permanent_address: Some(mac),
+            },
+        }
+    }
+}
+
+impl Display for WickedName {
```

Comment:
This `impl Display` is only necessary for config file naming.  Wicked is specific about it's config file names, so if using MAC, we need to remove the colons from the file name.